lib/jellyfishService.js

```diff
@@ -201,7 +201,7 @@ function createServer(serverConfig, mongoClient, seagullClient, userApiClient, g
         return cb({statusCode: 403, message: 'You do not have rights to view this data'});
       }
 
-      getPrivatePair(userId, function(err, privatePair) {
+      getPrivatePair(targetUserId, function(err, privatePair) {
         if (err) {
           log.warn(err, 'Failed to get private pair for user [%s]', userId);
```

Comment:
Yes, that was, in fact, the bug.  We need to get the privatePair for the *target* user since the data is stored and encrypted under the target user now (not the uploading user).